nixos/modules/services/security/vaultwarden/default.nix

```diff
@@ -196,6 +196,31 @@ in {
         AmbientCapabilities = "CAP_NET_BIND_SERVICE";
         StateDirectory = "bitwarden_rs";
         StateDirectoryMode = "0700";
+        RemoveIPC = true;
+        NoNewPrivileges = true;
+        CapabilityBoundingSet = "";
+        SystemCallFilter = [ "@system-service" ];
+        ReadWritePaths = [ "/var/lib/bitwarden_rs" ];
```

Comment:
```suggestion
        ReadWritePaths = [ cfg.config.DATA_FOLDER ];
```

Reply:
if this path should be set in `ReadWritePaths` this is certainly a better way to do it, thanks.
still unsure on if thats the case however given the stateDirectory setup

Reply:
`cfg.config.DATA_FOLDER` isn't always going to exist right? I for one don't set it.
doesn't setting it break this service in other ways? `configENV` seems to assume its the default path as does the backup service.

should I make a separate commit to attempt to address this?

maybe the [template env](https://github.com/dani-garcia/vaultwarden/blob/main/.env.template) file should also be linked in the `config` options description as well

for now I'm going to drop this line entirely as I'm not sure its required.

Reply:
> `cfg.config.DATA_FOLDER` isn't always going to exist right? I for one [don't set it](https://github.com/nu-nu-ko/crystal/blob/main/mods/services/vaultwarden.nix?rgh-link-date=2024-03-09T07%3A16%3A10Z#L23).

It should always exist https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/security/vaultwarden/default.nix#L28 https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/services/security/vaultwarden/default.nix#L206



> `configENV` seems to assume its the default path as does the backup service.

I want to go away from that because bitwarden_rs is no longer true and I already run it with the data directory being /var/lib/vaultwarden.

> should I make a separate commit to attempt to address this?

Please squash



> maybe the [template env](https://github.com/dani-garcia/vaultwarden/blob/main/.env.template?rgh-link-date=2024-03-09T07%3A16%3A10Z) file should also be linked in the `config` options description as well

yes, sure